### ISSUE TITLE
Pissible false-positive foreach with variable set to null at the end

### ISSUE THREAD

**Issue by avramcosmin** [kalessil/phpinspectionsea#782]
To reduce memory usage sometimes I do something like:

```
foreach($array as $item) {
   // doing something with $item
   // in case I don't use if($item) and if I'm setting $item = null after or inside foreach I get the bellow exception.
}
$item = null;
```

Each time I do this I receive the following: `NULL pointer exception may occur here.`

**Comment by kalessil** [comment]
Thank you for reporting @avramcosmin. Which the plugin and IDE versions is it?

**Comment by avramcosmin** [comment]
Thank you @kalessil for your great work.

PhpStorm latest (2017.3.4 Build: 173.4548.32). Same error was reported in all previous versions of PhpStorm. Plugin PHP Inspections (EA Extended) latest (after installing the latest PhpStorm version but the error was reported before as well).

[Here](https://github.com/kalessil/phpinspectionsea/blob/master/docs/probable-bugs.md#null-pointer-exceptions-prevention) is where I'm sent if I want to read more about why the exception was raised.

**Comment by kalessil** [comment]
You are welcome =) Ok, I'll try to reproduce it.

**Comment by kalessil** [comment]
Fixed
